### ISSUE TITLE
feat: implement Typography component

### ISSUE THREAD

**Issue by gustavopch** [nexxtway/react-rainbow#966]
-   [x] This component does not currently exist in the library.
-   [x] I have searched the [issues](https://github.com/nexxtway/react-rainbow/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior 🤔
It should be used to standardize all kinds of texts in an application (to be used by the lib consumers but also by the lib components themselves).

## Use cases 🌈
It should provide different variants (title, subtitle, body, button, caption), different colors (brand, textPrimary, textSecondary, success, danger, warning), alignment options (left, center, right, justify), maybe decoration options (underline, strike-through) etc.

Currently I think this lib doesn't provide theming, but this component should be made with theming in mind so that when theming is implemented one may be able to use Lato w400 for body but Raleway w700 for title.

## Context 🔦
Basically the same context as this:
https://material-ui.com/components/typography
https://material-ui.com/api/typography
